app/v3/service/internal/keeper/invocation.go

```diff
@@ -246,7 +246,9 @@ func (k Keeper) StartRequestContext(
 	requestContext.State = types.RUNNING
 	k.SetRequestContext(ctx, requestContextID, requestContext)
 
-	if requestContext.BatchState == types.BATCHCOMPLETED {
+	if requestContext.BatchState == types.BATCHCOMPLETED &&
```

Comment:
if the `NewRequestBatch` is the next batch, it seems better to remove it when pausing request context.

Reply:
My consideration: 
1. Removing needs another key
2. Low performance if repeated paused and started before timeout

Naturally deleted instead

Reply:
Add more comments for this code.
It's confused about
```golang
requestContext.BatchRequestCount == 0 || requestContext.BatchRequestCount != requestContext.BatchResponseCount
```

Reply:
or add a new batch state to describe this case.

Reply:
todo in https://github.com/irisnet/irishub/pull/2293